pinot-server/src/main/java/org/apache/pinot/server/starter/grpc/PinotQueryHandler.java

```diff
@@ -18,18 +18,258 @@
  */
 package org.apache.pinot.server.starter.grpc;
 
+import com.google.common.base.Preconditions;
 import io.grpc.stub.StreamObserver;
+import org.apache.commons.configuration.ConfigurationException;
+import org.apache.pinot.common.exception.QueryException;
+import org.apache.pinot.common.metrics.ServerMeter;
+import org.apache.pinot.common.metrics.ServerMetrics;
+import org.apache.pinot.common.metrics.ServerQueryPhase;
 import org.apache.pinot.common.proto.PinotQueryServerGrpc;
 import org.apache.pinot.common.proto.Server;
+import org.apache.pinot.common.utils.CommonConstants;
+import org.apache.pinot.common.utils.DataTable;
+import org.apache.pinot.core.common.datatable.DataTableImplV2;
+import org.apache.pinot.core.common.datatable.DataTableUtils;
+import org.apache.pinot.core.data.manager.InstanceDataManager;
+import org.apache.pinot.core.data.manager.SegmentDataManager;
+import org.apache.pinot.core.data.manager.TableDataManager;
+import org.apache.pinot.core.indexsegment.IndexSegment;
+import org.apache.pinot.core.indexsegment.mutable.MutableSegment;
+import org.apache.pinot.core.plan.Plan;
+import org.apache.pinot.core.plan.maker.InstancePlanMakerImplV2;
+import org.apache.pinot.core.plan.maker.PlanMaker;
+import org.apache.pinot.core.query.config.QueryExecutorConfig;
+import org.apache.pinot.core.query.exception.BadQueryRequestException;
+import org.apache.pinot.core.query.pruner.SegmentPrunerService;
+import org.apache.pinot.core.query.request.ServerQueryRequest;
+import org.apache.pinot.core.query.request.context.QueryContext;
+import org.apache.pinot.core.query.request.context.TimerContext;
+import org.apache.pinot.core.segment.index.metadata.SegmentMetadata;
+import org.apache.pinot.core.util.QueryOptions;
+import org.apache.pinot.core.util.trace.TraceContext;
+import org.apache.pinot.spi.env.PinotConfiguration;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.ExecutorService;
 
 /**
  * Handler for grpc server requests.
  * As data becomes available server responses will be added to the result stream.
  * Once the request is complete the client will aggregate the result metadata.
  */
 public class PinotQueryHandler extends PinotQueryServerGrpc.PinotQueryServerImplBase {
+  private static final Logger LOGGER = LoggerFactory.getLogger(PinotQueryHandler.class);
+
+  private InstanceDataManager _instanceDataManager = null;
+  private SegmentPrunerService _segmentPrunerService = null;
+  private PlanMaker _planMaker = null;
+  private long _defaultTimeOutMs = CommonConstants.Server.DEFAULT_QUERY_EXECUTOR_TIMEOUT_MS;
+  private ServerMetrics _serverMetrics;
+
+  public synchronized void init(PinotConfiguration config, InstanceDataManager instanceDataManager,
```

Comment:
It will be very hard to reuse the current `QueryScheduler` because that is not designed for streaming API. So for the first version we can just launch an `ExecutorService` within this class and use it to execute queries without introducing query scheduling.

Reply:
Makes sense, will be pushing an update for this.